src/pbft/pbft.h

```diff
@@ -221,7 +221,10 @@ namespace pbft
       {
         case pbft_message:
           serialized::skip(data, size, sizeof(PbftHeader));
-          message_receiver_base->receive_message((char*)((uint64_t)data), size);
+          if (message_receiver_base->pre_verify(data, size))
+          {
+            message_receiver_base->receive_message(data, size);
```

Comment:
I think it would be better to deal with the pre_verify inside the receive_message callback.

Reply:
ok, you mean have replica do the pre-verifying in this case? This PR can then be made redundant

Reply:
Yes. I think it would be better. How is (data,size) reclaimed? Do we need to make a copy inside the receive_message callback?

Reply:
yes we create a new Message and then delete it when we are done with it. See here: https://github.com/microsoft/ePBFT/pull/36/files#diff-fb85cd747a35e5be5a1a064eb8d9993dR356

Reply:
yes we create a new Message and then delete it when we are done with it. See here: https://github.com/microsoft/ePBFT/pull/36/files#diff-fb85cd747a35e5be5a1a064eb8d9993dR356

Reply:
yes we create a new Message and then delete it when we are done with it. See here: https://github.com/microsoft/ePBFT/pull/36/files#diff-fb85cd747a35e5be5a1a064eb8d9993dR356